src/routes/activityReports/handlers.test.js

```diff
@@ -35,171 +24,170 @@ const mockResponse = {
   })),
 };
 
-const reportObject = {
-  activityRecipientType: 'grantee',
-  status: 'draft',
-  userId: mockUser.id,
-  lastUpdatedById: mockUser.id,
-  resourcesUsed: 'test',
+const mockRequest = {
+  session: {
+    userId: 1,
+  },
 };
 
-describe('Activity Report handlers', () => {
-  let user;
-
-  beforeAll(async () => {
-    user = await User.create(mockUser, { include: [{ model: Permission, as: 'permissions' }] });
-  });
-
-  afterAll(async () => {
-    await ActivityRecipient.destroy({ where: {} });
-    await ActivityReport.destroy({ where: {} });
-    await User.destroy({ where: { id: user.id } });
-    db.sequelize.close();
-  });
+const report = {
+  id: 1,
+  resourcesUsed: 'resources',
+  userId: 1,
+};
 
+describe('Activity Report handlers', () => {
   afterEach(() => {
     jest.clearAllMocks();
   });
 
-  describe('getApprovers', () => {
-    const approverOne = {
-      id: 50,
-      name: 'region 5',
-      permissions: [
-        {
-          userId: 50,
-          regionId: 5,
-          scopeId: SCOPES.APPROVE_REPORTS,
-        },
-      ],
+  describe('submitReport', () => {
+    const request = {
+      ...mockRequest,
+      params: { activityReportId: 1 },
+      body: { approvingManagerId: 1, additionalNotes: 'notes' },
     };
-    const approverTwo = {
-      id: 51,
-      name: 'region 6',
-      permissions: [
-        {
-          userId: 51,
-          regionId: 6,
-          scopeId: SCOPES.APPROVE_REPORTS,
-        },
-      ],
+
+    it('returns the report', async () => {
+      ActivityReport.prototype.canUpdate = jest.fn().mockReturnValue(true);
```

Comment:
This is some of my javascript ignorance showing, but, how does this implementation impact the rest of the tests? Does this pollute everything? Or is it somehow scoped to this test?

Reply:
That is a good question, I'll look into it.